src/internalEnforcer.ts

```diff
@@ -89,10 +114,42 @@ export class InternalEnforcer extends CoreEnforcer {
     return ok;
   }
 
+  /**
+   *
+   * addPoliciesInternalWithoutNotify will add multiple policy rules to the current policy without calling watcher.
+   */
+  protected async addPoliciesInternalWithoutNotify(sec: string, ptype: string, rules: string[][]): Promise<boolean> {
+    for (const rule of rules) {
+      if (this.model.hasPolicy(sec, ptype, rule)) {
+        return false;
+      }
+    }
+
+    if (this.autoSave) {
+      if ('addPolicies' in this.adapter) {
+        try {
+          await this.adapter.addPolicies(sec, ptype, rules);
+        } catch (e) {
+          if (e.message !== 'not implemented') {
```

Comment:
This condition is pretty much "if `addPolicies` is not implemented, throw an error, but if it's not implemented do not throw an error".

Reply:
@Sefriol this behavior is all over Casbin, https://github.com/casbin/casbin/blob/3f721c48dfab03e84cbdbdec11bc1160c6166f55/internal_api.go#L237

@nodece should we throw error for not implemented?

Reply:
Usually, we should throw an error, but the golang-casbin didn't throw, so the node-casbin follows this behavior.

Let's keep this behavior. 

I have an idea here, we can add a flag to check this error, when the user wants to check this error, the user needs to enable this flag.

Reply:
@nodece IMO user can know by themselves from the docs of whichever watcher they are using whether that method exists or not, I don't think they really need to get this error.
But if that's not the case, then I think we should just show the error even if it's 'not implemented' so that even if the user is not aware they will at least know that reason through the error.

Reply:
Let's say if we add a flag for 'not implemented', there is very little chance that a method in the adapter will be removed in the next update and so the user will know it using that flag.

Reply:
This flag is used to throw errors. if the flag is true, we need to throw the `not implemented`.

Reply:
@nodece what do you think if we make a flag for all errors thrown by the adapter including `not implemented`.

Reply:
@nodece I have added changes for the above idea, the tests are failing rn but we will fix it once you aprove the idea.

Reply:
You can keep old implement, and then you can make a new PR to do that.